cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/SourceControlManager.java

```diff
@@ -0,0 +1,244 @@
+    if (config.getProvider() == Provider.GITHUB) {
+      if (config.getAuth().getType() == AuthType.PAT) {
+        return new GitPATAuthStrategy(this.context.getStore());
+      }
+    }
+    throw new AuthStrategyNotFoundException(String.format("No strategy found for provider %s and type %s.",
+                                                          config.getProvider(),
+                                                          config.getAuth().getType()));
+  }
+
+  /**
+   * Validated the configuration provided to this manager.
+   *
+   * @return a {@link List} of {@link  Exception}s occurred during validation.
+   */
+  public List<Exception> validateConfig() {
+    ArrayList<Exception> validationErrors = new ArrayList<>();
+    // Validate existence of secret in Secure Store.
+    RepositoryConfig config = context.getRepositoryConfig();
+    LOG.info("Checking existence of secret in secure store: " + config.getAuth().getTokenName());
+    try {
+      context.getStore().get(context.getNamespaceId().getNamespace(), config.getAuth().getTokenName()).get();
+    } catch (Exception e) {
+      validationErrors.add(new Exception("Failed to get password/token from secure store", e));
+      return validationErrors;
+    }
+
+    // Try fetching heads in the remote repository.
+    FileRepository localRepo = null;
+    if (validationRepoDirectory.exists() && !deleteDirectory(validationRepoDirectory)) {
+      validationErrors.add(new Exception(String.format("Failed to clean up local directory '%s' for temp repository",
+                                                       validationRepoDirectory.getAbsolutePath())));
+      return validationErrors;
+    }
+    try {
+      localRepo = new FileRepository(validationRepoDirectory);
+    } catch (IOException e) {
+      validationErrors.add(new Exception(String.format("Failed to create local git repository at path '%s'",
+                                                       validationRepoDirectory.getAbsolutePath()), e));
+    }
+
+    if (localRepo == null) {
+      return validationErrors;
+    }
+
+    Collection<Ref> refs = null;
+    LOG.info("Created local repo in directory for validation: " + localRepoDirectory.getAbsolutePath());
+    try (Git git = new Git(localRepo)) {
+      LsRemoteCommand cmd = git.lsRemote().setRemote(config.getLink()).setHeads(true).setTags(false);
+      cmd = getAuthenticatedCommandWithTimeout(cmd);
+      refs = cmd.call();
+    } catch (TransportException e) {
+      validationErrors.add(new Exception("Failed to connect with remote repository. " +
+                                           "Authentication credentials may be invalid or the repository may not exist.",
+                                         e));
+    } catch (Exception e) {
+      validationErrors.add(e);
+    }
+
+    // Check if default branch exists.
+    if (refs != null && !Strings.isNullOrEmpty(config.getDefaultBranch())) {
```

Comment:
Same to above comment, we need to consider the case that `config.getDefaultBranch()` returns null/empty (defaultBranch is optional)

Reply:
Added the check.